version: '3.8'

services:
  db:
    image: postgres:14.1-alpine
    container_name: postgres_test_cont
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    ports:
      - 5432:5432
    environment:
      - POSTGRES_DB=rose_test
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres

  web:
    build: .
    container_name: web_test_cont
    command: >
      bash -c "python manage.py migrate
      && python manage.py load_param_settings
      && python manage.py runserver 0.0.0.0:8005"
    volumes:
      - .:/code
      - /etc/oroose_test:/config
    ports:
      - "8005:8005"
    environment:
      - LOAD_CONFIG_PATH=/config/.env.test
      - MAIN_PROCESS=TESTING
    depends_on:
      - db

  cache:
    image: redis:6.2-alpine
    container_name: cache_test_cont
    ports:
      - '6379:6379'
    command: redis-server --save 20 1 --loglevel warning
    volumes:
      - cache:/data

  celery:
    build: .
    container_name: celery_test_cont
    command: celery -A oroose worker -l info
    volumes:
      - .:/code
      - /etc/oroose_test:/config
    environment:
      - LOAD_CONFIG_PATH=/config/.env.test
      - MAIN_PROCESS=TESTING
    depends_on:
      - db
      - cache

  celery-beat:
    build: .
    container_name: celery_beat_test_cont
    command: celery -A oroose beat -l info
    volumes:
      - .:/code
      - /etc/oroose_test:/config
    environment:
      - LOAD_CONFIG_PATH=/config/.env.test
      - MAIN_PROCESS=TESTING
    depends_on:
      - db
      - cache

volumes:
  postgres_data:
  cache: